bfm/src/arch/linux/ioctl.cpp

```diff
@@ -22,82 +22,72 @@
 #include <ioctl.h>
 #include <ioctl_private.h>
 
-ioctl::ioctl() noexcept
-{
-    m_d = std::make_shared<ioctl_private>();
-}
+ioctl::ioctl() noexcept :
+    m_d {std::make_unique<ioctl_private>()}
```

Comment:
I'm not to sure if this constructor should be marked `noexcept` as `std::make_unique` has potential to throw `std::bad_alloc` or if `ioctl_private`'s constructor throws will violate the `noexcept` contract...